bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,503 @@
+                    "Error communicating with Tibber API");
+            return;
+
+        } else {
+            logger.debug("Unexpected response from Tibber: {}", jsonResponse);
+            return;
+        }
+    }
+
+    public void getURLHourly(String url) throws IOException {
+        InputStream inputStream = priceInfo.getHourlyInputStream(configuration.getHomeid());
+        String jsonResponse = HttpUtil.executeUrl("POST", url, httpHeader, inputStream, null, REQUEST_TIMEOUT);
+        logger.debug("API response (Hourly): {}", jsonResponse);
+
+        if (!jsonResponse.contains("error") && jsonResponse.contains("cost")) {
+            try {
+                JsonObject Object = (JsonObject) new JsonParser().parse(jsonResponse);
+                JsonObject myObject = (JsonObject) Object.getAsJsonObject("data").getAsJsonObject("viewer")
+                        .getAsJsonObject("home").getAsJsonObject("hourly").getAsJsonArray("nodes").get(0);
+
+                String hourlyFrom = myObject.get("from").toString();
+                String hourlyTo = myObject.get("to").toString();
+                updateState(HOURLY_FROM, new StringType(hourlyFrom));
+                updateState(HOURLY_TO, new StringType(hourlyTo));
+
+                String cost = myObject.get("cost").toString();
+                updateChannel(HOURLY_COST, cost);
+
+                String consumption = myObject.get("consumption").toString();
+                updateChannel(HOURLY_CONSUMPTION, consumption);
+
+                return;
+
+            } catch (JsonSyntaxException e) {
+                logger.error("JsonException: {}", e);
+            }
+        } else if (jsonResponse.contains("error")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error communicating with Tibber API");
+            return;
+
+        } else {
+            logger.debug("Unexpected response from Tibber: {}", jsonResponse);
+            return;
+        }
+    }
+
+    public void startRefresh(int Refresh) {
+        if (pollingJob == null) {
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    updateRequest();
+                } catch (IOException e) {
+                    logger.error("IO Exception: {}", e);
+                }
+            }, 1, Refresh, TimeUnit.MINUTES);
+        }
+        return;
+    }
+
+    public void updateRequest() throws IOException {
+        getURLPriceInfo(BASE_URL);
+        getURLDaily(BASE_URL);
+        getURLHourly(BASE_URL);
+
+        if (rtEnabled.equals("true") && !isConnected()) {
+            try {
+                open();
+            } catch (Exception e) {
+                logger.error("Exception: {}", e);
+            }
+        }
+
+        return;
+    }
+
+    public void updateChannel(String channelID, String channelValue) {
+        if (!channelValue.contains("null")) {
```

Comment:
If it's null it should probably set the state to `UNDEF`.

Reply:
Uncertain if this would be the correct approach? 

The channel remains Online/initialised, but send actual 'null' values (e.g around hour and day transition). And for graphing purposes, the binding is set up to only update channel value if value received is not null.